src/batched/KokkosBatched_LU_Team_Internal.hpp

```diff
@@ -62,11 +59,16 @@ namespace KokkosBatched {
 
         if (tiny != 0) {
           if (member.team_rank() == 0) {
```

Comment:
For the `tiny != 0` case, the team's Thread 0 modifies the diagonal entry.  Is that access coherent with respect to the rest of the team?  The code that follows needs the (modified) diagonal entry on all threads in the team, not just Thread 0.

Reply:
the root thread took a reference of the diagonal value and modified it.

Reply:
@kyungjoo-kim Right, but then the other threads need to see that updated diagonal value.  Are write-to-read memory accesses across different threads in the same team always coherent?

Reply:
Probably not. I will move the barrier upwards before each thread greb the diagonal values.

Reply:
@mhoemmen thanks for pointing it out (it is quite subtle).

Reply:
Jup that barrier needs moving up

Reply:
@kyungjoo-kim Cool; I'll be happy to review again when ready :-D .